File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.go
func rearrangeArray(nums []int) []int {
    var temp = make([]int, len(nums))
    pos := 0
    neg := 1
    
    for i:=0;i<len(nums);i++ {
        if nums[i] > 0{
            temp[pos] = nums[i]
            pos+=2
        }else{
            temp[neg] = nums[i]
            neg+=2
        }
    }
    nums = temp
    return nums
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int temp[] = new int[nums.length];
        int pos = 0,neg = 1;
        for(int i=0;i<nums.length;i++){
            if(nums[i] > 0){
                temp[pos] = nums[i];
                pos=pos+2;
            }
            else{
                temp[neg] = nums[i];
                neg=neg+2;
            }
        }
        nums = temp;
        return nums;
    }
}